### ISSUE TITLE
Settings Sync sign in hangs - can't sign in (Ubuntu 20.04 & 22.04, Insiders and Stable)

### ISSUE THREAD

**Issue by maxkwallace** [microsoft/vscode#187674]
Type: <b>Bug</b>

See screen recordings I made of the issue here:

https://youtu.be/wZ5uyCMRBFI (22.04, Stable, includes video of Github Authentication output log)
https://youtu.be/iV3jOZxxsBg (20.04, Insiders)

I also tried with Insiders on 22.04, nothing works.

Backstory here is that I've been using VS Code Insiders for a few months on Ubuntu 20.04, working well, never had any issues. Just got a new PC. Did a fresh install of Ubuntu 22.04. Installed Insiders, tried to sign in, immediately ran into this issue. Tried removing (apt remove), reinstalling, tried with Stable instead of Insiders, tried with Chrome instead of Firefox. Tried with my Microsoft account instead of GitHub (just to see if I could authenticate at all-- I assume this wouldn't actually sync my saved settings) -- nothing worked.

Meanwhile I'm still signed in to Insiders on my old PC (Ubuntu 20.04). Signed out to see if I could replicate the issue there. Yup. Now I can't sign in on either PC.

Let me know if I can provide any additional info or replication recordings, very motivated to figure this out!

VS Code version: Code 1.80.0 (660393deaaa6d1996740ff4880f1bad43768c814, 2023-07-04T13:39:48.236Z)
OS version: Linux x64 5.19.0-46-generic
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i5-11400F @ 2.60GHz (12 x 3979)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 1, 1|
|Memory (System)|15.48GB (8.55GB free)|
|Process Argv|--unity-launch --crash-reporter-id fca3de0a-f721-4bed-8533-991fa166e073|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263cf:30784848
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
vscccc:30610679
2e4cg342:30602488
pyind779:30671433
89544117:30613380
pythonsymbol12:30671437
a9j8j154:30646983
showlangstatbar:30737416
vsctsb:30748421
pythonms35:30701012
03d35959:30757346
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30771522
hideindicator:30785051
pythongtdpath:30769146
i26e3531:30780429
pythonnosm12tcf:30779713
pythonidxpt:30784022
pythonnocebcf:30776496
h7j2d465:30786200
cmakesidepanelv2:30779593

```

</details>

**Comment by TylerLeonhardt** [comment]
In both videos, you stop before going through all possible auth flows. Cancelling before the one that May help the most, the device code flow.

Can you go through all possible flows? (The "would you like to try a different way?" Notification)

**Comment by maxkwallace** [comment]
@TylerLeonhardt thanks for the reply! I did go through those flows (clicking "Yes") last night-- and they didn't work either. You're correct, I should have included those in the recordings.

Anyway, I tried again just now and suddenly worked normally on the first try, first flow. I haven't even rebooted since last night. 

![Screenshot from 2023-07-12 09-00-01](https://github.com/microsoft/vscode/assets/3249377/5d164209-6e4d-482f-b7b4-cb709893b248)

![Screenshot from 2023-07-12 09-00-15](https://github.com/microsoft/vscode/assets/3249377/2ccc16b9-0f71-4fc7-9bc0-2afe5706efc1)

I didn't see either of these popups last night. Makes me think it's something on the GitHub web end. 

Anyway, I'll close this issue for now, thanks for the help!
